src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/OkHttpWebSocketWrapper.java

```diff
@@ -87,14 +89,29 @@ public void onMessage(WebSocket webSocket, String message) {
         @Override
         public void onClosing(WebSocket webSocket, int code, String reason) {
             onClose.invoke(code, reason);
-            closeSubject.onComplete();
+            try {
+                closeLock.lock();
+                closeSubject.onComplete();
```

Comment:
I'm assuming the user can't force us to take this lock in any strange ways so there's no chance of a dead lock.

Reply:
Right, the user has no control over the callsite to this lock.